### PR TITLE
Allow RT Validation to Use Most Recent Available Schedule Feeds (Up to 7 Days)

### LINKED ISSUE #2306: Bug: We're failing to validate RT data during UTC morning
**Describe the bug**
I was looking at Sentry and noticed we're failing to find schedule data for RT validation consistently in the morning.

**To Reproduce**
See [this issue](https://sentry.calitp.org/organizations/sentry/issues/36046/?project=2&query=is%3Aunresolved&sort=freq&statsPeriod=14d) which unfortunately has a misleading fingerprint, since it's the same exception but across various feeds.

I think this started occurring when we [moved the Schedule download later](https://github.com/cal-itp/data-infra/pull/1890), to account for Airtable download and config generation.

**Expected behavior**
We probably want to either look back to the prior day, or shift RT processing even later? Something we should discuss before rushing out a fix.

In addition, we should adjust the fingerprinting of this processing job to account for the feed so tasks are grouped more granularly.

**Additional context**
This is easily backfilled but is probably resulting in some slightly odd validation notice patterns. Notices are generally consumed at the day level in the warehouse, and anything with a counter greater than 0 fails the check.

### PR DESCRIPTION
**Pull request by SorenSpicknall**
# Description

We currently have a gap in RT validation caused by missing schedule extracts during times of day before the current day's schedule extracts have been processed. This PR allows RT validation to look back on the previously extracted version of the schedule, with a delay of up to seven days, before failing.

Resolves #2306

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

Various pieces of the changed code have been tested locally, but at the time of PR creation the whole GTFS RT Parser job has not yet been run in local Airflow.

### CODE REVIEW

**Review by atvaccaro — CHANGES_REQUESTED** [review]

File: jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py [hunk 1]
```diff
@@ -364,21 +368,35 @@ def validate_and_upload(
...
+                f"no schedule data found for {first_extract.path} and day {target_date}"
+            )
+            fallback_days += 1
+
+    if gtfs_zip is None:
         raise ScheduleDataNotFound(
-            f"no schedule data found for {first_extract.path}"
-        ) from e
+            f"no recent schedule data found for {first_extract.path}"
+        )
```
[the remark this answers is not in the document]
Comment (SorenSpicknall) [reply jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py]: Resolved in [b8f81e7](https://github.com/cal-itp/data-infra/pull/2400/commits/b8f81e79e8396547ff9cde872b6fcd606245fc9d)

File: jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py [hunk 2]
```diff
@@ -364,21 +368,35 @@ def validate_and_upload(
...
+    gtfs_zip = None
+    fallback_days = 0
+    while (
+        fallback_days < 7
+    ):  # Fall back to most recent available schedule within 7 days
+        try:
+            target_date = first_extract.dt - datetime.timedelta(days=fallback_days)
+            schedule_extract = get_schedule_extracts_for_day(target_date)[
+                first_extract.config.base64_validation_url
+            ]
```
[the remark this answers is not in the document]
Comment (SorenSpicknall) [reply jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py]: Resolved in [e17298d](https://github.com/cal-itp/data-infra/pull/2400/commits/e17298d73e45b88f7008f457e7c84ea34a5ecaac)

File: jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py [hunk 3]
```diff
@@ -364,21 +368,31 @@ def validate_and_upload(
...
-            first_extract.config.base64_validation_url
-        ]
-        gtfs_zip = download_gtfs_schedule_zip(
-            fs,
-            schedule_extract=schedule_extract,
-            dst_dir=tmp_dir,
-            pbar=pbar,
-        )
-    except (KeyError, FileNotFoundError) as e:
+    today = first_extract.dt
```
Comment (atvaccaro) [inline jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py:371]: nit: maybe `extract_day` or something

**Review by atvaccaro — APPROVED** [review]
lgtm!

**Pull request merged**